include/sacn/source.h

```diff
@@ -55,52 +55,125 @@ typedef int sacn_source_t;
+  bool send_unicast_only;
+
+  /*! If non-zero, this is the synchronization universe used to synchronize the sACN output. Defaults to 0. */
+  uint16_t sync_universe;
+
+} SacnSourceUniverseConfig;
+
+/*! A default-value initializer for an SacnSourceUniverseConfig struct. */
+#define SACN_SOURCE_UNIVERSE_CONFIG_DEFAULT_INIT \
+  {                                              \
+    0, NULL, 0, 100, NULL, 0, 0, 0               \
+  }
+
+void sacn_source_universe_config_init(SacnSourceUniverseConfig* config, uint16_t universe, const uint8_t* values_buffer,
+                                      size_t values_len, const uint8_t* priority_buffer);
+
+etcpal_error_t sacn_source_create(const SacnSourceConfig* config, sacn_source_t* handle, SacnNetworkChangeResult* good_interfaces);
+void sacn_source_destroy(sacn_source_t handle);
 
-etcpal_error_t sacn_source_change_priority(sacn_source_t handle, uint8_t new_priority);
-etcpal_error_t sacn_source_change_preview_flag(sacn_source_t handle, bool new_preview_flag);
 etcpal_error_t sacn_source_change_name(sacn_source_t handle, const char* new_name);
 
-void sacn_source_set_dirty(sacn_source_t handle, uint8_t start_code);
-void sacn_sources_set_dirty(const SacnSourceStartCodePair* start_codes, size_t num_start_codes);
-void sacn_source_send_now(sacn_source_t handle, uint8_t start_code);
-void sacn_sources_send_now(const SacnSourceStartCodePair* start_codes, size_t num_start_codes);
+etcpal_error_t sacn_source_add_universe(sacn_source_t handle, const SacnSourceUniverseConfig* config, bool dirty_now);
+void sacn_source_remove_universe(sacn_source_t handle, uint16_t universe);
+
+etcpal_error_t sacn_source_add_unicast_destination(sacn_source_t handle, uint16_t universe, const EtcPalIpAddr* dest,
+                                                   bool dirty_now);
+void sacn_source_remove_unicast_destination(sacn_source_t handle, uint16_t universe, const EtcPalIpAddr* dest);
+
+etcpal_error_t sacn_source_change_priority(sacn_source_t handle, uint16_t universe, uint8_t new_priority);
+etcpal_error_t sacn_source_change_preview_flag(sacn_source_t handle, uint16_t universe, bool new_preview_flag);
+etcpal_error_t sacn_source_change_synchronization_universe(sacn_source_t handle, uint16_t universe,
+                                                           uint16_t new_sync_universe);
+
+etcpal_error_t sacn_source_send_now(sacn_source_t handle, uint16_t universe, uint8_t start_code, const uint8_t* buffer,
+                                    size_t buflen);
+etcpal_error_t sacn_source_send_synchronization(sacn_source_t handle, uint16_t universe);
+
+void sacn_source_set_dirty(sacn_source_t handle, uint16_t universe);
+void sacn_source_set_list_dirty(sacn_source_t handle, const uint16_t* universes, size_t num_universes);
+void sacn_source_set_dirty_and_force_sync(sacn_source_t handle, uint16_t universe);
+
+size_t sacn_source_process_sources(void);
```

Comment:
I agree.  I still tend to use size_t when I don't want a negative value, but I agree with the sentiment in the style guide.  My main concern is to avoid signed/unsigned mismatches, so I tend to skew with whatever is reasonable.  I'll change it here.